common/src/main/java/org/wildfly/httpclient/common/HttpStickinessHelper.java

```diff
@@ -0,0 +1,356 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.httpclient.common;
+
+import io.undertow.client.ClientRequest;
+import io.undertow.client.ClientResponse;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.handlers.Cookie;
+import io.undertow.server.handlers.CookieImpl;
+import io.undertow.util.Cookies;
+import io.undertow.util.HeaderMap;
+import io.undertow.util.HeaderValues;
+import io.undertow.util.Headers;
+import io.undertow.util.HttpString;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Helper methods for processing JSESSIONID Cookies and Headers used in stickiness processing.
+ *
+ * @author <a href="mailto@rachmato@redhat.com>Richard Achmatowicz</a>
+ */
+public class HttpStickinessHelper {
+
+    private static final HttpString STRICT_STICKINESS_HOST = new HttpString("StrictStickinessHost");
+    private static final HttpString STRICT_STICKINESS_RESULT = new HttpString("StrictStickinessResult");
+    private static final HttpString JSESSIONID_COOKIE_NAME = new HttpString("JSESSIONID");
```

Comment:
Keep in mind, the logic in this class is specific to Apache httpd-based load balancers.
Most other load balancers (e.g. HAProxy) use a distinct cookie for routing.  The name of the cookie should be configurable.  In this case, there is no need for a "sessionID" cookie, as the SFSB ID is already available via an HTTP header.